src/SourceLocator/Type/AnonymousClassObjectSourceLocator.php

```diff
@@ -101,7 +100,7 @@ public function __construct(private string $fileName, private int $startLine)
              */
             public function enterNode(Node $node)
             {
-                if (! ($node instanceof Node\Stmt\Class_) || $node->name !== null) {
+                if (! ($node instanceof Node\Stmt\Class_) || $node->name !== null || $node->getLine() !== $this->startLine) {
```

Comment:
Would it be better to check start column too? Multiple classes on the same line can happen, especially in malicious software (which BetterReflection should be able to analyse)

Reply:
Ah, I see that this code throws on multiple same-line classes 😩